### PR TITLE
Fix the null check when converting to PSRoledefinition

### PR DESCRIPTION
**Pull request by veseshad92**
## Description

Checking if the whole permissions object is null and handle it respectively. 

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x ] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [x ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Resources/Resources/Models.Authorization/AuthorizationClientExtensions.cs [hunk 1]
```diff
@@ -58,8 +58,8 @@ public static PSRoleDefinition ToPSRoleDefinition(this RoleDefinition role)
                     AssignableScopes = role.AssignableScopes.ToList(),
                     Description = role.Description,
                     IsCustom = role.RoleType == CustomRole ? true : false,
-                    Condition = role.Permissions[0]?.Condition,
-                    ConditionVersion = role.Permissions[0]?.ConditionVersion
+                    Condition = role.Permissions == null ? null : role.Permissions[0]?.Condition,
+                    ConditionVersion = role.Permissions == null ? null : role.Permissions[0]?.ConditionVersion
```
Comment (BethanyZhou) [inline src/Resources/Resources/Models.Authorization/AuthorizationClientExtensions.cs:62]:
```suggestion
                    Condition = role?.Permissions?[0]?.Condition,
                    ConditionVersion = role?.Permissions?[0]?.ConditionVersion
```

File: src/Resources/Resources/ChangeLog.md [hunk 2]
```diff
@@ -22,6 +22,7 @@
 * Introduced secrets detection feature to safeguard sensitive data.
 * Migrated SDK generation from autorest csharp to autorest powershell.
 * Fixed deployment and deployment stack New/Set cmdlets to fail if template/parameter uri fails to downloads.
+* Added null check to the permissions object in the ToPSRoleDefinition method to return if the whole permissions object array is null.
```
Comment (BethanyZhou) [inline src/Resources/Resources/ChangeLog.md:25]: Please insert your change log between Line21-22 to avoid missing new changes.

**Comment by NoriZC** [comment]
Please share the daily build to https://github.com/Azure/azure-powershell/issues/24177 after merge

File: src/Resources/Resources/ChangeLog.md (hunk 2 above)
Comment (veseshad92) [reply src/Resources/Resources/ChangeLog.md]: updated

**Comment by BethanyZhou** [comment]
@veseshad92 please resolve conflicts in ChangeLog.md.

**Comment by veseshad92** [comment]
> @veseshad92 please resolve conflicts in ChangeLog.md.

updated :)

**Review by BethanyZhou — APPROVED** [review]
LGTM

**Pull request merged**